martian/src/metadata.rs

```diff
@@ -182,39 +178,32 @@ impl Metadata {
     }
 
     /// Update the Martian journal -- so that Martian knows what we've updated
-    fn update_journal_main(&mut self, name: &str, force: bool) -> Result<()> {
+    fn update_journal(&self, name: &str) -> Result<()> {
         let journal_name: Cow<str> = if self.stage_type != "main" {
             format!("{}_{name}", self.stage_type).into()
         } else {
             name.into()
         };
 
-        if force || !self.cache.contains(name) {
-            let tmp_run_file = format!("{}.{journal_name}.tmp", self.run_file);
-            let run_file = &tmp_run_file[..tmp_run_file.len() - 4];
-
-            {
-                let mut f = File::create(&tmp_run_file)?;
-                if let Err(err) = f.write_all(make_timestamp_now().as_bytes()) {
-                    // Pretty much ignore this error.  The only reason we need
-                    // any content at all in this file is because some
-                    // filesystems behave strangely with completely empty files.
-                    eprintln!("Writing journal file {tmp_run_file}: {err}");
-                }
+        let tmp_run_file = format!("{}.{journal_name}.tmp", self.run_file);
+        let run_file = &tmp_run_file[..tmp_run_file.len() - 4];
+
+        {
+            let mut f = File::create(&tmp_run_file)?;
+            if let Err(err) = f.write_all(make_timestamp_now().as_bytes()) {
+                // Pretty much ignore this error.  The only reason we need
+                // any content at all in this file is because some
+                // filesystems behave strangely with completely empty files.
+                eprintln!("Writing journal file {tmp_run_file}: {err}");
             }
-            rename(tmp_run_file.as_str(), run_file).or_else(ignore_not_found)?;
-            self.cache.insert(journal_name.into_owned());
```

Comment:
Not sure why we had this cache in the first place but it wasn't serving any real purpose.  The only metadata files we might write to repeatedly are
1. `_log`, which is handled by `mrjob`
2. `_progress`, which this isn't exposing and for which we'd want to write the journal unconditionally every time.
3. `_alarms`, for which `mrp` doesn't actually need the journal at all. And if you're writing enough alarms that journaling it repeatedly causes performance degradation, you have other problems to worry about.